addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1905,6 +1905,63 @@ options.registry.collapse = options.Class.extend({
     },
 });
 
+options.registry.MasonryLayout = options.Class.extend({
+    xmlDependencies: (options.Class.xmlDependencies || []).concat(
+        ['/website/static/src/snippets/s_masonry_block/s_masonry_block.xml']
+    ),
+
+    /**
+     * @override
+     */
+    start: function () {
+        if (!this.$target[0].dataset.currentTemplate) {
+            this.$target[0].dataset.currentTemplate = "default";
+            const $template = $(qweb.render(`website.s_template_masonry.default`, {}));
+            this.$('.container, .container-fluid, .o_container_small').append($template);
+        }
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the masonry snippet layout with templates.
+     *
+     * @see this.selectClass for parameters
+     */
+    editLayout: function (previewMode, widgetValue, params) {
+        this.$target[0].dataset.currentTemplate = widgetValue;
+        const $container = this.$('.container, .container-fluid, .o_container_small');
+        const $template = $(qweb.render(`website.s_template_masonry.${widgetValue}`, {}));
```

Comment:
`widgetValue` should contain the whole name so that:
- Grepping the whole name of the template allow to find all of its occurences (we should avoid building template names)
- An external module or theme can add other templates to the list without having to name them in a specific way